File: j2k/src/org/jetbrains/jet/j2k/ast/DoWhileStatement.java
package org.jetbrains.jet.j2k.ast;

import org.jetbrains.annotations.NotNull;

/**
 * @author ignatov
 */
public class DoWhileStatement extends WhileStatement {
  public DoWhileStatement(Expression condition, Statement statement) {
    super(condition, statement);
  }

  @NotNull
  @Override
  public String toKotlin() {
    return "do" + N +
      myStatement.toKotlin() + N +
      "while" + SPACE + "(" + myCondition.toKotlin() + ")";
  }
}

File: src/org/jetbrains/jet/j2k/ast/DoWhileStatement.kt
package org.jetbrains.jet.j2k.ast


public open class DoWhileStatement(condition: Expression, statement: Statement): WhileStatement(condition, statement) {
    public override fun toKotlin(): String {
        return "do\n" + statement.toKotlin() + "\nwhile (" + condition.toKotlin() + ")"
    }
}
